packages/amplify-provider-awscloudformation/lib/system-config-manager.js

```diff
@@ -54,76 +55,107 @@ function setProfile(awsConfig, profileName) {
   fs.writeFileSync(configFilePath, ini.stringify(config));
 }
 
-function getProfile(profileName) {
+async function getProfiledAwsConfig(profileName, isRoleSourceProfile) {
   let awsConfig;
-  if (fs.existsSync(credentialsFilePath) && fs.existsSync(configFilePath)) {
-    let accessKeyId;
-    let secretAccessKey;
-    let region;
+  const profileConfig = getProfileConfig(profileName);
+  if (profileConfig) {
+    if (!isRoleSourceProfile && profileConfig.role_arn && profileConfig.source_profile) {
+      const roleCredentials =
+        await getRoleCredentials(profileConfig);
+      delete profileConfig.role_arn;
+      delete profileConfig.source_profile;
+      awsConfig = {
+        ...profileConfig,
+        ...roleCredentials,
+      };
+    } else {
```

Comment:
Not sure as to in which case this branch would get hit?

Reply:
it is hit for regular profile that does not use roles.